force-app/main/default/classes/services/GauExpenditure/tests/GauExpenditureSelector_TEST.cls

```diff
@@ -0,0 +1,686 @@
+        System.assertNotEquals(
+            null,
+            disbursementsWhenNonDisbursementId,
+            'GauExpenditureSelector.getDisbursementsById should never return null.'
+        );
+
+        System.assert(
+            disbursementsWhenNonDisbursementId.isEmpty(),
+            'GauExpenditureSelector.getDisbursementsById should return empty when disbursementId does not belong to a non-deleted outfunds__Disbursement__c record.'
+        );
+
+        // When disbursementId DOES belong to a non-deleted outfunds__Disbursement__c record.
+        System.assertNotEquals(
+            null,
+            actual,
+            'GauExpenditureSelector.getDisbursementsById should never return null.'
+        );
+
+        System.assertEquals(
+            1,
+            actual.size(),
+            'GauExpenditureSelector.getDisbursementsById should return one record.'
+        );
+
+        // Assert Disbursement with disbursementId was queried and Name, Amount, and Status fields are queried.
+        System.assertEquals(disbursementId, actual[0].Id);
+        System.assertNotEquals(null, actual[0].Name);
+        System.assertEquals(1000, actual[0].outfunds__Amount__c);
+        System.assertEquals('Scheduled', actual[0].outfunds__Status__c);
+
+        System.assertEquals(3, actual[0].GAU_Expendatures__r.size());
+
+        Map<String, GAU_Expenditure__c> expendituresByGauName = new Map<String, GAU_Expenditure__c>();
+        for (GAU_Expenditure__c expenditure : actual[0].GAU_Expendatures__r) {
+            expendituresByGauName.put(
+                expenditure.General_Accounting_Unit__r.Name,
+                expenditure
+            );
+        }
+
+        {
+            final GAU_Expenditure__c expenditure = expendituresByGauName.get('GAU 0');
+
+            System.assertNotEquals(null, expenditure);
+            System.assertEquals(
+                true,
+                expenditure.General_Accounting_Unit__r.npsp__Active__c
+            );
+            System.assertEquals(700, expenditure.Amount__c);
+        }
+
+        {
+            final GAU_Expenditure__c expenditure = expendituresByGauName.get('GAU 1');
+
+            System.assertNotEquals(null, expenditure);
+            System.assertEquals(
+                false,
+                expenditure.General_Accounting_Unit__r.npsp__Active__c
+            );
+            System.assertEquals(200, expenditure.Amount__c);
+        }
+
+        {
+            final GAU_Expenditure__c expenditure = expendituresByGauName.get('GAU 2');
+
+            System.assertNotEquals(null, expenditure);
+            System.assertEquals(
+                true,
+                expenditure.General_Accounting_Unit__r.npsp__Active__c
+            );
+            System.assertEquals(100, expenditure.Amount__c);
+        }
+    }
+
+    @IsTest
+    private static void getDisbursementsById_MissingCrud() {
```

Comment:
Yes, I wanted to test this but ran out of time.  There's some super non-obvious challenges testing permissions:
- It seems fields from other packages grant FLS to all of its packaged fields.  So, testing FLS is super hard for selectors on other package objects.
    - e.g. I tried deleting `FieldPermissions` for all permission sets related to the Test User (including it's Profile) for `outfunds__Disbursement__c` but all fields were **still*** accessible to the Test User!  (i.e. `Schema.outfunds__Disbursement__c.SObjectType.fields.outfunds__Amount__c.getDescribe().isAccessible() == true`).  It may be a bug where Apex tests don't faithfully represent production permission behavior.  I know of one other place where Apex tests don't faithfully represent production permission behavior: compound fields.
- Permissions are slightly complicated for objects with a master-detail relationship.